File: Java/deepspace/src/deepspace/PowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 * Esta clase representa ....
 * @authors Manuel Diaz-Meco Terrés y Ana Graciani Donaire
 */
public class PowerEfficientSpaceStation extends SpaceStation {
    private final static float EFFICIENCYFACTOR = 1.10f;
    
    PowerEfficientSpaceStation(SpaceStation station) {
        super(station);
    }
    
    @Override
    public float fire() {
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection() {
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot l) {
        Transformation t = super.setLoot(l);
        if (t == Transformation.SPACECITY)
            return Transformation.NOTRANSFORM;
        else
            return t;        
    }
    
    PowerEfficientSpaceStationToUI getToUI() {
        return new PowerEfficientSpaceStationToUI(this);
    }
    
}

File: Ruby/PowerEfficientSpaceStation.rb
#encoding: UTF-8

# Autores: Manuel Díaz-Meco Terrés y Ana Graciani Donaire
# 2ºDGIIM		PDOO

# Esta clase representa el concepto de estación espacial 
# eficiente así como sus atributos y funcionalidades

require_relative 'SpaceStation'
#require_relative 'Transformation'
require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace
	class PowerEfficientSpaceStation < SpaceStation
	
		private
	
		@@EFFICIENCYFACTOR = 1.10

		public
		
		def initialize(spacestation)
			copy(spacestation)
		end
		
		def fire
			return super*@@EFFICIENCYFACTOR
		end
		
		def protection
			return super*@@EFFICIENCYFACTOR
		end
		
		def setLoot(loot)
			trans = super

            if trans == Transformation::SPACECITY 
               	return Transformation::NOTRANSFORM
            else
            	return trans
            end
		end
		
		def getUIversion
		      	return PowerEfficientSpaceStationToUI.new(self)
		end
	end
end
